msticpy/data/azure_sentinel.py

```diff
@@ -0,0 +1,209 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Uses the Azure Python SDK to collect and return details related to Azure."""
+from typing import Dict, List
+
+import pandas as pd
+import requests
+from azure.common.exceptions import CloudError
+
+from .azure_data import AzureData
+
+_PATH_MAPPING = {
+    "ops_path": "/providers/Microsoft.SecurityInsights/operations",
+    "alert_rules": "/providers/Microsoft.SecurityInsights/alertRules",
+    "ss_path": "/savedSearches",
+    "bookmarks": "/providers/Microsoft.SecurityInsights/bookmarks",
+}
+
+
+class AzureSentinel(AzureData):
+    """Class for reutrning key Azure Sentinel elements."""
+
+    def __init__(self, connect: bool = False):
+        """Initialize connector for Azure APIs."""
+        super().__init__()
+
+    def connect(
+        self,
+        auth_methods: List = None,
+        silent: bool = False,
+    ):
+        """Authenticate with the SDK & API."""
+        super().connect(auth_methods=auth_methods, silent=silent)
+        self.token = _get_token(self.credentials)
+        self.res_group_url = None
+        self.prov_path = None
+
+    def get_sentinel_workspaces(self, sub_id: str) -> Dict:
+        """
+        Return a list of Azure Sentinel workspaces in a Subscription.
+
+        Parameters
+        ----------
+        sub_id : str
+            The subscription ID to get a list of workspaces from
+
+        Returns
+        -------
+        Dict
+            A dictionary of workspace names and ids
+
+        """
+        print("Finding Azure Sentinel Workspaces...")
+        res = self.get_resources(sub_id=sub_id)
+        # handle no results
+        if isinstance(res, pd.DataFrame) and not res.empty:
+            sentinel = res[
+                (res["resource_type"] == "Microsoft.OperationsManagement/solutions")
+                & (res["name"].str.startswith("SecurityInsights"))
+            ]
+            workspaces = []
+            for wrkspace in sentinel["resource_id"]:
+                res_details = self.get_resource_details(
+                    sub_id=sub_id, resource_id=wrkspace
+                )
+                workspaces.append(res_details["properties"]["workspaceResourceId"])
+
+            workspaces_dict = {}
+            for wrkspace in workspaces:
+                name = wrkspace.split("/")[-1]
+                workspaces_dict.update({name: wrkspace})
+
+            return workspaces_dict
+
+        print(f"No Azure Sentinel workspaces in {sub_id}")
+        return {}
+
+    def get_hunting_queries(self, res_id) -> pd.DataFrame:
```

Comment:
Surprised that you didn't get any warnings about these being instance methods but not referencing self.

Reply:
They reference self.token (and with the updates I've added self._check_config as well now)